eslint.config.js

```diff
@@ -1,44 +1,41 @@
-import { fixupConfigRules } from '@eslint/compat';
-import pluginJs from '@eslint/js';
-import * as reactQuery from '@tanstack/eslint-plugin-query';
-import pluginImport from 'eslint-plugin-import';
-import pluginReactConfig from 'eslint-plugin-react/configs/recommended.js';
-import * as reactHooks from 'eslint-plugin-react-hooks';
 import globals from 'globals';
-import tseslint from 'typescript-eslint';
+import eslint from '@eslint/js';
+import ts from 'typescript-eslint';
+import reactRefresh from 'eslint-plugin-react-refresh';
+import reactHooks from 'eslint-plugin-react-hooks';
+import * as tanstackQuery from '@tanstack/eslint-plugin-query';
 
 export default [
-  { languageOptions: { globals: globals.browser } },
-  pluginJs.configs.recommended,
-  ...tseslint.configs.recommended,
-  ...fixupConfigRules(pluginReactConfig),
+  eslint.configs.recommended,
+  ...ts.configs.recommended,
+  {
+    ignores: ['dist/**/*', 'node_modules/**/*', 'eslint.config.js'],
+  },
+  {
+    files: ['*.ts', '*.tsx'],
+    languageOptions: {
+      ecmaVersion: 2022,
```

Comment:
Why `2022`?

https://eslint.org/docs/latest/use/configure/language-options#specifying-javascript-options

They say:
> In most cases, we recommend using the default of "latest" to ensure you’re always using the most recent ECMAScript version.

Reply:
In `eslint.cjs` config we had 2020. So i update to 2022. 
And if you use latest. There could be an issue where lint starts failing, because new ecmaScript was introduced. 

I prefer to change such thing manually and solve all issues during such update. 


But if you want i can change to the latest. It is up to you 🙃

Reply:
Let's leave it as it is for now. Maybe we'll update it in the next "iteration of refactorization".

Reply:
Sure :)

Reply:
in `tsconfig.json` we have `"lib": ["ES2020", "DOM", "DOM.Iterable", "ES2023"],`. So we can set 2023

Reply:
Set to 2023